src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -235,8 +260,54 @@ static term start_sta(term sta_config, GlobalContext *global)
             return BADARG_ATOM;
         }
     }
+    char *hostname = driver_data->hostname;
+    if (term_is_invalid_term(hostname_term)) {
+        size_t buf_size = DEFAULT_HOSTNAME_SIZE;
+        hostname = malloc(buf_size);
+        if (IS_NULL_PTR(hostname)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            term error = get_default_device_name(hostname, global);
+            if (error != OK_ATOM) {
+                free(ssid);
+                free(psk);
+                free(hostname);
+                return error;
+            }
+        }
+    } else {
+        int ok = 0;
+        char *buf = malloc(MAX_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            buf = interop_term_to_string(hostname_term, &ok);
+        }
+        if (!ok || IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return BADARG_ATOM;
+        }
+        hostname = malloc(strlen(buf) + 1);
```

Comment:
I am confused about this allocation. When is it freed?

Reply:
see above.